### PR TITLE
Add skywalking log exporter

### PR DESCRIPTION
**Pull request by liqiangz**
Description: Add skywalking log exporter -- First PR

Link to tracking Issue: apache/skywalking#6209

Testing:

logdata_to_logrecord_test.go

Documentation:
exporter/skywalkingexporter/README.md

### CODE REVIEW

File: exporter/skywalkingexporter/README.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
+# skywalking gRPC Exporter
+
+Exports data via gRPC using [skywalking-data-collect-protocol)](
```
Comment (jpkrohling) [inline exporter/skywalkingexporter/README.md:3]:
```suggestion
Exports data via gRPC using [skywalking-data-collect-protocol](
```

File: exporter/skywalkingexporter/README.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+https://github.com/apache/skywalking-data-collect-protocol)
+format. By default, this exporter requires TLS and offers queued retry capabilities.
+
+Supported pipeline types: logs
+
+## Getting Started
+
+The following settings are required:
+
+- `endpoint` (no default): host:port to which the exporter is going to send skywalking log data,
```
Comment (jpkrohling) [inline exporter/skywalkingexporter/README.md:13]:
```suggestion
- `endpoint` (no default): host:port to which the exporter is going to send SkyWalking log data,
```

All the mentions to the project name should use the right capitalization.

File: exporter/skywalkingexporter/README.md [hunk 3]
```diff
@@ -0,0 +1,45 @@
+# skywalking gRPC Exporter
```
Comment (jpkrohling) [inline exporter/skywalkingexporter/README.md:1]:
```suggestion
# SkyWalking gRPC Exporter
```

File: exporter/skywalkingexporter/README.md [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+- `key_file` (no default): path to the TLS key to use for TLS required connections. Should
+  only be used if `insecure` is set to false.
+
+Example:
+
+```yaml
+exporters:
+  skywalking:
+    endpoint: "1.2.3.4:11800"
+    Insecure: true
```
Comment (jpkrohling) [inline exporter/skywalkingexporter/README.md:36]:
```suggestion
    insecure: true
```

File: exporter/skywalkingexporter/go.mod [hunk 5]
```diff
@@ -0,0 +1,11 @@
+module github.com/open-telemetry/opentelemetry-collector-contrib/exporter/alibabacloudlogserviceexporter
+
+go 1.16
```
Comment (jpkrohling) [inline exporter/skywalkingexporter/go.mod:3]: Please, double-check this. I think we moved to 1.17 for this repo already.

File: exporter/skywalkingexporter/logdata_to_logrecord.go [hunk 6]
```diff
@@ -0,0 +1,163 @@
...
+	severityNumber         = "severityNumber"
+	severityText           = "severityText"
+	name                   = "name"
+	flags                  = "flags"
+	instrumentationName    = "otlp.name"
+	instrumentationVersion = "otlp.version"
+	defaultServiceName     = "otel-collector"
+)
+
+func logDataToLogRecode(ld pdata.Logs) []*logpb.LogData {
```
Comment (jpkrohling) [inline exporter/skywalkingexporter/logdata_to_logrecord.go:37]: Please, follow the contribution guidelines for new components, specifically the part about breaking the work into multiple PRs.

**Comment by liqiangz** [comment]
Updated, Could you please have a review on this.  @jpkrohling

**Review by jpkrohling — APPROVED** [review]

**Comment by liqiangz** [comment]
Could you please trigger the ci.  @jpkrohling

File: exporter/skywalkingexporter/config.go [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+
+// Config defines configuration for SkyWalking exporter.
+type Config struct {
+	config.ExporterSettings       `mapstructure:",squash"` // squash ensures fields are correctly decoded in embedded struct
+	configgrpc.GRPCClientSettings `mapstructure:",squash"` // squash ensures fields are correctly decoded in embedded struct.
+	exporterhelper.QueueSettings  `mapstructure:"sending_queue"`
+	exporterhelper.RetrySettings  `mapstructure:"retry_on_failure"`
+
+	// The number of workers that send the gRPC requests.
+	NumWorkers int `mapstructure:"num_workers"`
```
Comment (bogdandrutu) [inline exporter/skywalkingexporter/config.go:31]: There is one in the QueueSettings

**Comment by liqiangz** [comment]
@bogdandrutu Updated. Could you please have a review again.

File: exporter/skywalkingexporter/config_test.go [hunk 8]
```diff
@@ -0,0 +1,87 @@
...
+	factories.Exporters[typeStr] = factory
+	cfg, err := configtest.LoadConfigAndValidate(path.Join(".", "testdata", "config.yaml"), factories)
+
+	require.NoError(t, err)
+	require.NotNil(t, cfg)
+
+	e0 := cfg.Exporters[config.NewID(typeStr)]
+	assert.Equal(t, e0, factory.CreateDefaultConfig())
+
+	e1 := cfg.Exporters[config.NewIDWithName(typeStr, "2")]
```
Comment (bogdandrutu) [inline exporter/skywalkingexporter/config_test.go:46]: Please rebase and fix breaking changes related to some function names.

**Review by bogdandrutu — APPROVED** [review]

**Comment by liqiangz** [comment]
@bogdandrutu Updated.

**Comment by tigrannajaryan** [comment]
@liqiangz please add yourself and anyone else who you believe can help maintain this component to https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/.github/CODEOWNERS

**Review by tigrannajaryan — APPROVED** [review]
LGTM, can be merged once codeowners is updated.

**Comment by liqiangz** [comment]
> @liqiangz please add yourself and anyone else who you believe can help maintain this component to https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/.github/CODEOWNERS

@tigrannajaryan  Updated.

**Review by tigrannajaryan — APPROVED** [review]
LGTM.

**Pull request merged**